### ISSUE TITLE
'ignore' option doesn't work in pylint 4.0.0 anymore

### ISSUE THREAD

**Issue by marmarek** [pylint-dev/pylint#10669]
### Bug description

The ignore option seems to be ignored now.
Files layout:

```
[user@host pylint-ignore]$ tree -s
[       4096]  .
└── [       4096]  module
    ├── [          0]  __init__.py
    ├── [         77]  main.py
    └── [       4096]  test
        └── [         46]  __init__.py

3 directories, 3 files
[user@host pylint-ignore]$ cat module/main.py 
# pylint: disable=too-few-public-methods,missing-docstring
class A:
    pass
[user@host pylint-ignore]$ cat module/test/__init__.py 

class i:
    """many issues here"""
    pass
```

### Configuration

```ini
[main]
ignore=test
```

### Command used

```shell
pylint module
```

### Pylint output

```python
************* Module module.test
module/test/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module/test/__init__.py:4:4: W0107: Unnecessary pass statement (unnecessary-pass)
module/test/__init__.py:2:0: R0903: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 2.50/10 (previous run: 2.50/10, +0.00)
```

### Expected behavior

`module/test` directory should be ignored. pylint 3.3.7 output:

```
------------------------------------
Your code has been rated at 10.00/10
```

### Pylint version

```shell
pylint 4.0.1
astroid 4.0.1
Python 3.13.7 (main, Aug 14 2025, 00:00:00) [GCC 14.3.1 20250523 (Red Hat 14.3.1-1)]

same issue with pylint 4.0.0 too
```

### OS / Environment

Fedora 41

### Additional dependencies

```python

```

**Comment by RemiCardona** [comment]
got this too, I have `ignore=migrations` in my many django projects to ignore autogenerated migration files.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas @jacobtylerwalls Anything that could explain this regression?

**Comment by Pierre-Sassoulas** [comment]
I have no idea this is pretty surprising to me, but the number of thumbsups doesn't lie. I suggest bissecting using a reproducer to find the culprit, I wasn't able to look into it yet.

**Comment by correctmost** [comment]
I was able to reproduce the issue with this folder structure:

```
.
├── __init__.py
├── a.py
├── migrations
│   ├── __init__.py
│   └── a.py
└── pyproject.toml
```

**a.py**
```python
# pass
```

**migrations/a.py**
```python
x
```

**pyproject.toml**
```toml
[tool.pylint.main]
ignore = "migrations"
```

The bisection led to commit abb81878fd114f.

**Comment by jacobtylerwalls** [comment]
@temyurchenko would you have any availability to look into this?

**Comment by temyurchenko** [comment]
> [@temyurchenko](https://github.com/temyurchenko) would you have any availability to look into this?

I'll try to look over the weekends

**Comment by sil-lnagel** [comment]
For us the issue was resolved after upgrading to 4.0.2, so maybe this was fixed accidentally?

**Comment by gthiemonge** [comment]
> For us the issue was resolved after upgrading to 4.0.2, so maybe this was fixed accidentally?

I can still reproduce the issue with 4.0.2 here

**Comment by RemiCardona** [comment]
I had to block 4.0.2 too, still an issue on my end.

**Comment by jacobtylerwalls** [comment]
@Pierre-Sassoulas we should just revert https://github.com/pylint-dev/pylint/commit/abb81878fd114f309297b35e1d392bee0d196fbf if feasible.

**Comment by Pierre-Sassoulas** [comment]
Let's do this, and @temyurchenko can do the actual fix without pressure in a minor

**Comment by shifqu** [comment]
Hello. In the commit we would like to revert, a change is made to not pass an ignore_list, but rather an empty list. I don't know exactly WHY this change was made, but by reverting that bit to pass `ignore_list` again, my pylint functions properly again.

I created a PR #10720 for this, but once again, I am not sure why the empty list is passed, so this might intorduce some regression. I guess @temyurchenko, as the author of the commit, is best to review this.
